r/R/main.R

```diff
@@ -60,19 +59,21 @@ calc_K <- function(k, TC=25, S=35, Mg=0.0528171, Ca=0.0102821, P=NULL, MyAMI_cal
   }
   
   # Calculate correction factor with MyAMI
-  if(MyAMI_calc == TRUE){ 
-    Fcorr = pymyami$calc_Fcorr(Sal=S, TempC=TC, Mg=Mg, Ca=Ca)
-  } else {
-    Fcorr = pymyami$approximate_Fcorr(Sal=S, TempC=TC, Mg=Mg, Ca=Ca)
-  }
+  if(run_MyAMI == TRUE){
+    if(MyAMI_calc == TRUE){ 
+      Fcorr = pymyami$calc_Fcorr(Sal=S, TempC=TC, Mg=Mg, Ca=Ca)
+    } else {
+      Fcorr = pymyami$approximate_Fcorr(Sal=S, TempC=TC, Mg=Mg, Ca=Ca)
+    }
   
-  # Apply correction
-  KF = Fcorr[[k]]
-  if(!is.null(KF)){
-    check_KF = ifelse(KF != 0, KF, 1)
-    K = K * check_KF
+    # Apply correction
+    KF = Fcorr[[k]]
+    if(!is.null(KF)){
+      check_KF = ifelse(KF != 0, KF, 1)
```

Comment:
I'm also not sure this line is necessary... pymyami should never produce a case where KF == 0... I think?

Reply:
True, that is a remain from an earlier error tracing. Removed in new PR.